runtime/compiler/z/codegen/S390J9CallSnippet.cpp

```diff
@@ -527,20 +527,45 @@ TR::S390VirtualUnresolvedSnippet::emitSnippetBody()
    *(uintptrj_t *) cursor = (uintptrj_t) callNode->getSymbolReference()->getCPIndexForVM();
    cursor += TR::Compiler->om.sizeofReferenceAddress();
 
+   // instruction to be patched
    *(uintptrj_t *) cursor = (uintptrj_t) (getPatchVftInstruction()->getBinaryEncoding());
    AOTcgDiag1(comp, "add TR_AbsoluteMethodAddress cursor=%x\n", cursor);
    cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor, NULL, TR_AbsoluteMethodAddress, cg()),
                           __FILE__, __LINE__, callNode);
    cursor += sizeof(uintptrj_t);
 
+   // Field used by nestmate private calls
+   // J9Method pointer of the callee. Initialized to 0.
+   *(uintptrj_t *) cursor = 0;
+   cursor += sizeof(uintptrj_t);
+
+   // Field used by nestmate private calls
+   // J2I thunk address
+   // TODO: leave out the AOT until AOT supports this.
+   *(uintptrj_t *) cursor = (uintptrj_t) thunkAddress;
+   // AOTcgDiag1(comp, "add ThunkAddress cursor=%x\n", cursor);
+   // cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor, NULL, TR_AbsoluteMethodAddress, cg()),
+   //                        __FILE__, __LINE__, callNode);
+   cursor += sizeof(uintptrj_t);
+
+   // Field used by nestmate private calls
+   // For private functions, this is the return address.
+   // Add a 2 bytes off set because it's the instruction after the BASR, which is 2-byte long
```

Comment:
Could you please add a fatal assert that `getIndirectCallInstruction() == TR::InstOpCode::BASR` since we are making the assumption here? Alternatively can't we use `getIndirectCallInstruction()->getBinaryLength()` instead of hardcoding 2 here and avoid adding the assert?